xarray/backends/api.py

```diff
@@ -175,10 +176,14 @@ def _get_backend_cls(engine):
 
 
 def _normalize_path(path):
-    if is_remote_uri(path):
-        return path
-    else:
-        return os.path.abspath(os.path.expanduser(path))
+    if isinstance(path, Path):
+        path = str(path)
+
+    if isinstance(path, str):
+        if not is_remote_uri(path):
+            path = os.path.abspath(os.path.expanduser(path))
```

Comment:
This can be simplified a bit...
```suggestion
    if isinstance(path, str) and not is_remote_uri(path):
        path = os.path.abspath(os.path.expanduser(path))
```

Reply:
D'oh!